### ISSUE TITLE
Don't use `System.exit(1)`

### ISSUE THREAD

**Issue by ewels** [nf-core/tools#1840]
We've been giving @pditommaso nightmares with our use of `System.exit`:

https://github.com/nf-core/tools/blob/bc56cdd35077e408a0a7c90ec07fcdff7c2d1f3b/nf_core/pipeline-template/lib/WorkflowPipeline.groovy#L16-L17

```console
$ nextflow run .
N E X T F L O W  ~  version 22.08.2-edge
Launching `./main.nf` [crazy_swirles] DSL2 - revision: 6b60f60cc2
Genome fasta file not specified with e.g. '--fasta genome.fa' or via a detectable config file.
```

Instead, we should be using exceptions:

```nextflow
throw new Exception("Genome fasta file not specified with e.g. '--fasta genome.fa' or via a detectable config file.")
```

```console
$ nextflow run .
N E X T F L O W  ~  version 22.08.2-edge
Launching `./main.nf` [cranky_roentgen] DSL2 - revision: 62f715c40c
Genome fasta file not specified with e.g. '--fasta genome.fa' or via a detectable config file.

 -- Check script 'main.nf' at line: 1 or see '.nextflow.log' file for more details
```

### TODO

- [ ] Remove from the template
- [ ] Add lint tests that search all files for `System.exit`

**Comment by ewels** [comment]
An easier method is to use `error` which is [a wrapper](https://github.com/nextflow-io/nextflow/blob/f38fd2db57baa98249c9cd9454a29bdae22e7f2c/modules/nextflow/src/main/groovy/nextflow/Nextflow.groovy#L257-L263) around calling an exception.

```nextflow
error("Genome fasta file not specified with e.g. '--fasta genome.fa' or via a detectable config file.")
```

**Comment by awgymer** [comment]
Cannot push my code right now due to some GitHub issue but I had a go at writing some lint code:
```
╭─ [!] 6 Pipeline Test Warnings 
│ system_exit: System.exit in WorkflowNanoporeqc.groovy: System.exit(1)                                                                                                                            
│ system_exit: System.exit in WorkflowNanoporeqc.groovy: System.exit(1)                                                                                                                            
│ system_exit: System.exit in NfcoreSchema.groovy: System.exit(1)                                                                                                                                  
│ system_exit: System.exit in WorkflowMain.groovy: System.exit(0)                                                                                                                                  
│ system_exit: System.exit in WorkflowMain.groovy: System.exit(0)                                                                                                                                  
│ system_exit: System.exit in WorkflowMain.groovy: System.exit(1)                                                                                                                                  
```

So I think this should inform the template changes to make too.

**Comment by awgymer** [comment]
Closed in #2211
